api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -114,4 +116,23 @@ public Person updateFHIRPerson(Person thePerson, String theId) {
 			return createFHIRPerson(thePerson);
 		}
 	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.PersonService#retirePerson(String)
+	 */
+	@Override
+	public void retirePerson(String id) throws ResourceNotFoundException, NotModifiedException {
+		org.openmrs.Person person = Context.getPersonService().getPersonByUuid(id);
+		if (person == null) throw new ResourceNotFoundException(String.format("Person with id '%s' not found", id));
+		if (person.isPersonVoided()) return;
+		try {
+			person.setVoided(true);
```

Comment:
Sure. But the underlying API do nothing. See org.openmrs.api.PersonService#voidPerson(org.openmrs.Person, java.lang.String):
```Java
public Person voidPerson(Person person, String reason) throws APIException {
		if (person == null)
			return 
		return dao.savePerson(person);
}
```
I guess some logic was planned here but never implemented. In 1.12 implementation is the same. So it is sorta work around to make it working before correct person voiding is implemented.